tests/core/version/test_production_version_migration.py

```diff
@@ -0,0 +1,264 @@
+# Copyright 2023 Avaiga Private Limited
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with
+# the License. You may obtain a copy of the License at
+#
+#        http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software distributed under the License is distributed on
+# an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the License for the
+# specific language governing permissions and limitations under the License.
+
+import multiprocessing
+from unittest.mock import patch
+
+from src.taipy.core import Core, taipy
+from src.taipy.core.data._data_manager import _DataManager
+from src.taipy.core.scenario._scenario_manager import _ScenarioManager
+from taipy.config.common.scope import Scope
+from taipy.config.config import Config
+
+from ...conftest import init_config
+
+
+def twice(a):
+    return a * 2
+
+
+def triple(a):
+    return a * 3
+
+
+def migrate_pickle_path(dn):
+    dn.data_node_properties["path"] = "bar.pkl"
```

Comment:
This usage is not optimal. The user would prefer
```python
dn.path = "bar.pkl"
```

However, since path is not an attribute of the DataNodeModel, it will not work.